src/rdkafka_broker.h

```diff
@@ -198,6 +205,30 @@ struct rd_kafka_broker_s { /* rd_kafka_broker_t */
                                                    * if enabled. */
         rd_interval_t       rkb_connect_intvl;    /* Reconnect throttling */
 
+        /**< Persistent connection demand is tracked by
+         *   an counter for each type of demand.
+         *   The broker thread will maintain a persistent connection
+         *   if any of the counters are non-zero, and revert to
+         *   on-demand mode when they all reach zero.
+         *   After incrementing any of the counters a broker wakeup
+         *   should be signalled to expedite handling. */
+        struct {
+                /**< Producer: partitions are being produced to.
+                 *   Consumer: partitions are being fetched from.
+                 *
+                 *   Counter is maintained by the broker handler thread
+                 *   itself, no need for atomic/locking.
+                 *   Maybe be reset to 0 on each producer|consumer_serve() loop
```

Comment:
'Maybe be ...' has me completely confused.

Reply:
Mission accomplished